isic-archive-tasks/isic_archive_tasks/image.py

```diff
@@ -0,0 +1,199 @@
+            'parentType': 'item',
+            'parentId': imageId,
+            'name': '%s_superpixels_v%s.png' % (image['name'], SUPERPIXEL_VERSION),
+            'size': len(superpixelsEncodedStream.getvalue())
+        }, data=superpixelsEncodedStream.getvalue(), headers={
+            'Girder-Token': self.token
+        })
+        uploadSuperpixelsResponse.raise_for_status()
+
+        superpixelsFile = File().load(uploadSuperpixelsResponse.json()['_id'], force=True)
+        image = Image().load(imageId, force=True, exc=False)
+
+        superpixelsFileNameMatch = re.match(
+            r'^%s_superpixels_v([0-9.]+)\.png' % image['name'],
+            superpixelsFile['name'])
+        if not superpixelsFileNameMatch:
+            return
+
+        superpixelsVersion = float(superpixelsFileNameMatch.group(1))
+        superpixelsFile['superpixelVersion'] = superpixelsVersion
+        # Work around an upstream Girder bug where "File().validate" sets
+        #   superpixelsFile['exts'] = ['0', 'png']
+        superpixelsFile = File().validate(superpixelsFile)
+        superpixelsFile['exts'] = ['png']
+        superpixelsFile = File().save(superpixelsFile, validate=False)
+
+        image['superpixelsId'] = superpixelsFile['_id']
+        image['ingestionState']['superpixel-mask'] = True
+        Image().save(image)
+    except Exception:
+        sentry_sdk.capture_exception()
+        image['ingestionState']['superpixel-mask'] = False
+        Image().save(image)
+
+
+@app.task(bind=True, base=CredentialedGirderTask)
+def generateLargeImage(self, imageId):
+    from girder.plugins.isic_archive.models.image import Image
+    from girder.plugins.large_image.models.image_item import ImageItem
+    try:
+        imageItem = ImageItem().load(id=imageId, force=True)
+        imageFile = Image().originalFile(imageItem)
+        imageItem['largeImage'] = {
+            'sourceName': 'tiff',
+            'originalId': imageFile['_id'],
+        }
+        ImageItem().save(imageItem)
+
+        # todo url
+        originalFileStreamResponse = requests.get(
+            'http://isic-archive.test/api/v1/file/%s/download' % imageFile['_id'], headers={
+                'Girder-Token': self.token
+            }, stream=True)
+        originalFileStreamResponse.raise_for_status()
+
+        with tempfile.NamedTemporaryFile() as input_file:
+            input_file.write(originalFileStreamResponse.content)
+            input_file.flush()  # necessary
+
+            with tempfile.NamedTemporaryFile() as output_file:
+                vips_tiffsave(input_file.name, output_file.name)
+
+                newFilename = os.path.splitext(imageFile['name'])[0] + time.strftime(
+                    '%Y%m%d-%H%M%S') + '.tiff'
+                uploadLargeImageResp = requests.post('http://isic-archive.test/api/v1/file',
+                                                     params={
```

Comment:
Can you fix the formatting? Drop the first argument down to a new indentation level, and then you can move everything else left.